src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -397,7 +413,7 @@ void IcebergMetadata::updateSnapshot()
         configuration_ptr->getPath());
 }
 
-void IcebergMetadata::updateState(const ContextPtr & local_context)
+void IcebergMetadata::updateState(const ContextPtr & local_context, bool metadata_file_changed)
```

Comment:
```suggestion
void IcebergMetadata::updateState(const ContextPtr & local_context, bool metadata_version_changed)
```
(just a suggestion, previous name is also ok, just seems `metadata_version_changed` is a bit more correct)